apix/v1alpha2/inferencepool_conversion.go

```diff
@@ -19,119 +19,100 @@ package v1alpha2
 import (
 	"fmt"
 
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	runtime "k8s.io/apimachinery/pkg/runtime"
 
 	v1 "sigs.k8s.io/gateway-api-inference-extension/api/v1"
 )
 
 // ConvertTo converts this InferencePool (v1alpha2) to the v1 version.
-func (src *InferencePool) ConvertTo() (*v1.InferencePool, error) {
-	if src == nil {
-		return nil, nil
+func (src *InferencePool) ConvertTo(dst *v1.InferencePool) error {
+	if dst == nil {
+		return fmt.Errorf("dst cannot be nil")
 	}
-
-	v1Extension, err := convertEndpointPickerConfToV1(&src.Spec.EndpointPickerConfig)
+	v1Extension, err := convertExtensionRefToV1(src.Spec.ExtensionRef)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	v1Status, err := converStatusToV1(src.Status)
+	v1Status, err := convertStatusToV1(&src.Status)
 	if err != nil {
-		return nil, err
-	}
-	dst := &v1.InferencePool{
-		TypeMeta:   src.TypeMeta,
-		ObjectMeta: src.ObjectMeta,
-		Spec: v1.InferencePoolSpec{
-			TargetPortNumber: src.Spec.TargetPortNumber,
-			ExtensionRef:     v1Extension,
-		},
-		Status: *v1Status,
+		return err
 	}
+	dst.TypeMeta = src.TypeMeta
+	dst.ObjectMeta = src.ObjectMeta
+	dst.Spec.TargetPortNumber = src.Spec.TargetPortNumber
+	dst.Spec.ExtensionRef = v1Extension
+	dst.Status = *v1Status
 	if src.Spec.Selector != nil {
 		dst.Spec.Selector.MatchLabels = make(map[v1.LabelKey]v1.LabelValue, len(src.Spec.Selector))
 		for k, v := range src.Spec.Selector {
 			dst.Spec.Selector.MatchLabels[v1.LabelKey(k)] = v1.LabelValue(v)
 		}
 	}
-	return dst, nil
+	return nil
 }
 
 // ConvertFrom converts from the v1 version to this version (v1alpha2).
-func ConvertFrom(src *v1.InferencePool) (*InferencePool, error) {
+func (dst *InferencePool) ConvertFrom(src *v1.InferencePool) error {
```

Comment:
Change the signature to match the convention in k8s https://book.kubebuilder.io/multiversion-tutorial/conversion